File: demo-android/app/src/main/java/com/example/demo/ui/components/TaskChips.kt
package com.example.demo.ui.components

import androidx.compose.material3.AssistChip
import androidx.compose.material3.AssistChipDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import com.example.demo.data.model.TaskPriority
import com.example.demo.data.model.TaskStatus
import com.example.demo.ui.i18n.taskPriorityLabel
import com.example.demo.ui.i18n.taskStatusLabel
import com.example.demo.ui.theme.DoneGreen
import com.example.demo.ui.theme.HighRed
import com.example.demo.ui.theme.InProgressOrange
import com.example.demo.ui.theme.LowGreen
import com.example.demo.ui.theme.MediumOrange
import com.example.demo.ui.theme.TodoBlue

@Composable
fun StatusChip(status: TaskStatus, modifier: Modifier = Modifier) {
    val (label, color) = when (status) {
        TaskStatus.TODO -> taskStatusLabel(status) to TodoBlue
        TaskStatus.IN_PROGRESS -> taskStatusLabel(status) to InProgressOrange
        TaskStatus.DONE -> taskStatusLabel(status) to DoneGreen
    }
    TaskChip(label = label, color = color, modifier = modifier)
}

@Composable
fun PriorityChip(priority: TaskPriority, modifier: Modifier = Modifier) {
    val (label, color) = when (priority) {
        TaskPriority.LOW -> taskPriorityLabel(priority) to LowGreen
        TaskPriority.MEDIUM -> taskPriorityLabel(priority) to MediumOrange
        TaskPriority.HIGH -> taskPriorityLabel(priority) to HighRed
    }
    TaskChip(label = label, color = color, modifier = modifier)
}

@Composable
private fun TaskChip(label: String, color: Color, modifier: Modifier = Modifier) {
    AssistChip(
        onClick = {},
        enabled = false,
        label = { Text(label, style = MaterialTheme.typography.labelMedium) },
        modifier = modifier,
        colors = AssistChipDefaults.assistChipColors(
            disabledContainerColor = color.copy(alpha = 0.15f),
            disabledLabelColor = color
        )
    )
}

File: demo-ios/Demo/UI/Components/TaskChips.swift
import SwiftUI

struct StatusChip: View {
    private var locale: AppLocale { AppLocale.shared }
    let status: TaskStatus

    private var color: Color {
        switch status {
        case .todo: TaskColors.todoBlue
        case .inProgress: TaskColors.inProgressOrange
        case .done: TaskColors.doneGreen
        }
    }

    var body: some View {
        chipLabel(TaskLabels.statusLabel(for: status, locale: locale), color: color)
            .testTag("status-tag-\(status.rawValue)")
    }
}

struct PriorityChip: View {
    private var locale: AppLocale { AppLocale.shared }
    let priority: TaskPriority

    private var color: Color {
        switch priority {
        case .low: TaskColors.lowGreen
        case .medium: TaskColors.mediumOrange
        case .high: TaskColors.highRed
        }
    }

    var body: some View {
        chipLabel(TaskLabels.priorityLabel(for: priority, locale: locale), color: color)
            .testTag("priority-tag-\(priority.rawValue)")
    }
}

private func chipLabel(_ text: String, color: Color) -> some View {
    Text(text)
        .font(.caption.weight(.medium))
        .foregroundStyle(color)
        .padding(.horizontal, 10)
        .padding(.vertical, 4)
        .background(color.opacity(0.15))
        .clipShape(Capsule())
}
